include/range/v3/utility/box.hpp

```diff
@@ -191,45 +249,73 @@ namespace ranges
             }
         };
 
-        template<typename Element, typename Tag = Element, bool Inherit = true>
-        using box_if =
-            box<Element, Tag, Inherit && std::is_empty<Element>::value &&
-                                         !detail::is_final<Element>::value>;
+        template<typename Element, typename Tag>
+        class box<Element, Tag, detail::box_compress::coalesce>
```

Comment:
When `Element` is final, or in the - I imagine unlikely in real-world code - eventuality that some class with `n` bytes of non-empty representation multiply inherits the same empty base more than `n` times.

I think the marginal utility is low.